hawtio-system/src/main/java/io/hawt/web/UploadServlet.java

```diff
@@ -143,4 +147,8 @@ public void update(long pBytesRead, long pContentLength, int pItems) {
         return uploadedFiles;
     }
 
+    private boolean fileAllowed(FileItem fileItem) throws IOException {
+        FileUploadFilter uploadFilter = new FileUploadFilter();
+        return uploadFilter.accept(IOUtils.toByteArray(fileItem.getInputStream()));
```

Comment:
It seems that the whole content of an uploaded file is loaded to memory. What if the size of a file is huge (e.g. 1GB)?  Doesn't that cause OOM error or performance degradation?

Reply:
Fixed